OBW: Replace Extras step with Recommended

v18 opened the pull request:
**Please do not merge until the to do checkboxes are checked**

This PR replaces the Extras step with the Recommend step, which looks like this:

<img width="851" alt="new-recommend-step" src="https://user-images.githubusercontent.com/11487924/38774947-e15bee0e-4041-11e8-8627-ad6533649d6a.png">

It will show these possible items (if the user can install plugins/themes):

- **Storefront**, if the theme is not already installed, and if another WC-compatible theme is not already installed
- **Automated Taxes**, if the user's store country is in the EU, US, CA, or AU
- **Google Analytics**
- **MailChimp**

It will show a different blurb at the top. If all the options will be displayed, it will say:

> Select from the list below to enable automated taxes, Google Analytics, and MailChimp’s best-in-class email services — and design your store with our official, free WooCommerce theme.

If not all four options will be displayed, it will say:

> Enhance your store with these recommended features.

You can test it at `wp-admin/index.php?page=wc-setup&step=recommended`

# Design Differences
There are some differences between the proposed design and the final. Heres' the proposed design:

<img width="823" alt="recommend-step-design" src="https://user-images.githubusercontent.com/11487924/38774967-5d329776-4042-11e8-8c91-32ca6be949c5.png">

The main difference is that the checkboxes are not WC-stylized. I took a look, and we use the normal checkboxes in all other places in the onboarding wizard, so I used the same style here. @jameskoster, what do you think?

~~I also was not able to get the SVGs for storefront & automated taxes from your [original link](https://cloudup.com/ch3sYP1tCZe), @/jameskoster, can you re-upload them please?~~

Nevermind - I do have the SVGs!

# Visual reference

Here's what the old Extras step looked like:

<img width="942" alt="old-extras-step" src="https://user-images.githubusercontent.com/11487924/38774964-509ba0c0-4042-11e8-9f0b-cfdec0f69131.png">

# Todo
- [x] checkboxes different than design, is that cool? (purple vs plain)
- [x] use correct SVG for automated taxes
- [x] use correct  SVG for storefront
- [x] blurb differences cool?
- [x] do we need bigger mention of WooCommerce Services for automated taxes?
- [x] does anything need to be updated for the fix in #19728?

v18 committed:
OBW: Rename Extras Step as Recommended

v18 committed:
OBW: Update Storefront description and style

v18 committed:
OBW: Update automated taxes option

v18 committed:
OBW: Add MailChimp to recommended step

v18 committed:
OBW: Update Recommended step description

dechov commented:
Looks good, just a few notes and questions:

- On the store setup page there's a new stylized checkbox (the [code](https://github.com/woocommerce/woocommerce/blob/fcb74a30ff27c8ec16c7ab24ec6a60af557168b6/assets/css/wc-setup.scss) for which we probably want to generalize): <img width="661" alt="screen shot 2018-04-15 at 12 34 20 pm" src="https://user-images.githubusercontent.com/1867547/38780752-5cabf290-40a9-11e8-8b32-399f2a23c60f.png">

- Within recommended items, paragraph line spacing can be reduced to match design (`1.5em` I think) and avoid the second line appearing farther away than the heading

- Do we want the theme name in italic to match design? Also, do we still want a Storefront link somewhere, despite the modified copy?

- Do we want the wide "Continue" button from the design, or leave as is? Not specific to this step

- Is it right that only "Automated Taxes" should be disabled by default, despite all being "Recommended"? I think so, but not sure

- The plugins aren't being activated due to two issues I'm just uncovering, one of which I introduced in https://github.com/woocommerce/woocommerce/pull/19200 and the other of which is due to the MailChimp plugin's PHP file name not matching the repo slug — easiest way to fix the latter is probably by passing `'file'` alongside `'name'` and `'repo_slug'` and making a change in WC_Install to look at `'file'` if it's supplied. I'll try and address these ASAP

- checktextdomain found a few things:
```
includes/admin/class-wc-admin-setup-wizard.php
[L1593]  esc_attr_e  Missing text domain
[L1625]  esc_html_e  Missing text domain
[L1627]  esc_html_e  Incorrect text domain used ("woocoomerce")
Warning: 3 problems Use --force to continue.
```

jameskoster commented:
If we can pull in the checkbox styling as @dechov suggests that would be good, but it's not crucial.

> Also, do we still want a Storefront link somewhere, despite the modified copy?

Yeah I think we can link the "Storefront" text in the description.

> Is it right that only "Automated Taxes" should be disabled by default, despite all being "Recommended"? I think so, but not sure

That's a good point. I think it would look better if the only disabled item appeared last in the list as it's more situational.

Finally, I think we could do with a little more space between the list items.

v18 commented:
Thanks for the review @dechov and @jameskoster!

**Stylized checkboxes**

> On the store setup page there's a new stylized checkbox (the code for which we probably want to generalize)

> If we can pull in the checkbox styling as @dechov suggests that would be good, but it's not crucial.

Thanks for the tip. I took a look, and the previous code is relying on the `label` and `input` not being nested. It's going to take a bit of time to generalize the CSS, so I'm going to put that off to another PR.

**Paragraph line spacing**

> Within recommended items, paragraph line spacing can be reduced to match design (1.5em I think) and avoid the second line appearing farther away than the heading

Great find, fixed.

**Blurb stylin'**

> Do we want the theme name in italic to match design? Also, do we still want a Storefront link somewhere, despite the modified copy?

> Yeah I think we can link the "Storefront" text in the description.

Fixed both of these 👍 

**Continue button**

> Do we want the wide "Continue" button from the design, or leave as is? Not specific to this step

I don't think we should change the button, since this wasn't a OBW-wide re-design. I kept it the way it looks like in all the other OBW screens. cc @jameskoster for an opinion if I missed something. Here are screenshots showing it's the same as the current one:

<img width="876" alt="recommended-screen-with-continue-button" src="https://user-images.githubusercontent.com/11487924/38850008-64b7b822-41dd-11e8-9aff-405e809dde0b.png">
<img width="828" alt="shipping-screen-with-continue-button" src="https://user-images.githubusercontent.com/11487924/38850009-64d6b38a-41dd-11e8-93a1-d171a869d90d.png">

**Automated Taxes options is checked by default**

> Is it right that only "Automated Taxes" should be disabled by default, despite all being "Recommended"? I think so, but not sure
> That's a good point. I think it would look better if the only disabled item appeared last in the list as it's more situational.

Automated Taxes are not disabled by default, sorry my screenshot was a bit misleading. All items are checked by default. Automated Taxes will be disabled only if a user disables them, goes to the next step, and then clicks the back button to go back to the Recommended step.

**Activated automatically bug**

> The plugins aren't being activated due to two issues
> easiest way to fix the latter is probably by passing 'file' alongside 'name' and 'repo_slug' and making a change in WC_Install to look at 'file' if it's supplied.

@dechov is there something you need me to do in this PR?

**Translation issues**

> checktextdomain found a few things:

Thanks, fixed.

**Spacing between items**

> Finally, I think we could do with a little more space between the list items.

Updated to 1.5em. Also removed the extra spacing that was showing up for the last item.

**Removing Google Analytics**

As discussed in p7bje6-S6-p2, I removed Google Analytics from the list.

Here's the final result:

<img width="839" alt="screen shot 2018-04-17 at 3 44 15 am" src="https://user-images.githubusercontent.com/11487924/38855513-a88d138a-41f1-11e8-8383-fb7b52ab3a55.png">

v18 commented on includes/admin/class-wc-admin-setup-wizard.php:
```diff
@@ -121,6 +121,22 @@ protected function should_show_automated_tax_extra() {
+		return current_user_can( 'install_plugins' );
+	}
+
+	protected function should_show_recommended_step() {
```
Thanks @claudiulodro - I fixed the issues, but I don't see that Travis CI is run again on this PR. Is there an issue / can you point me to the right spot? Thanks!

dechov commented:
> Automated Taxes are not disabled by default

I'm not sure how the default was originally decided and have no input on what it should be, but AFAICT it's unchanged from 3.3 which defaults to disabled. Tested this branch on a fresh install:

<img width="655" alt="screen shot 2018-04-17 at 10 40 13 am" src="https://user-images.githubusercontent.com/1867547/38877831-88275dd0-422d-11e8-87b8-fd321b4fa4ab.png">

If the design indicates an intentional change to this, we can probably just set `true` as `get_option` default.

> is there something you need me to do in this PR?

This line would make things compatible with https://github.com/woocommerce/woocommerce/pull/19728 (and not break anything without it):
```diff
@@ -1714,6 +1714,7 @@ class WC_Admin_Setup_Wizard {
                                array(
                                        'name'      => __( 'MailChimp for WooCommerce', 'woocommerce' ),
                                        'repo-slug' => 'mailchimp-for-woocommerce',
+                                       'file'      => 'mailchimp-woocommerce.php',
                                )
                        );
                }
```

claudiulodro reviewed (approved):
Looks good to me. Just needs merge conflicts resolved and that line added to make it compatible with #19728.

EDIT: Easiest way to resolve css file merge conflicts is to just delete the files with conflicts in them and run `grunt css` to generate new css files.

v18 commented:
@dechov - fixed the automated taxes issue by removing the option check. We're not checking for the other items (storefront, mailchimp) anyway, and we're generally not designing the OBW to be back-button compatible.

@claudiulodro thanks! Updated the CSS, and updated the plugin file to make it compatible with #19728. Good to go 👍
